gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -616,8 +618,13 @@ private ManagedChannel createSingleChannel() throws IOException {
     boolean useDirectPathXds = false;
     if (canUseDirectPath()) {
       CallCredentials callCreds = MoreCallCredentials.from(credentials);
+      // altsCallCredentials may be null and will be correctly ignored by
+      // GoogleDefaultChannelCredentials.
       ChannelCredentials channelCreds =
-          GoogleDefaultChannelCredentials.newBuilder().callCredentials(callCreds).build();
+          GoogleDefaultChannelCredentials.newBuilder()
+              .callCredentials(callCreds)
+              .altsCallCredentials(altsCallCredentials)
```

Comment:
if altsCallCredentials is non-null, do we need/ should pass both CallCredentials over? Is that the intended behavior for GoogleDefaultChannelCredentials?

i.e.
check null altscallcredentials -> only send callCreds
check non-null altscallcredentials -> only send altsCallCredentials

Reply:
Yes, GoogleDefaultChannelCredentials need the callCreds even when altsCallCreds is not null. Because gRPC internally allows DirectPath to CloudPath fallback. If the actual connection turns out to be CloudPath, the altsCallCreds cannot be used.